tensorflow/core/kernels/mkl/onednn_nn_ops_benchmark.cc

```diff
@@ -0,0 +1,158 @@
+  }                                                                          \
+  WRAP_BENCHMARK(                                                            \
+      JOIN(BM_oneDNN_Softmax_##dtype##_intraop_##num_threads##_dims,         \
+           JOIN(CONCAT_DIMS, PP_NARG(__VA_ARGS__))(__VA_ARGS__)))            \
+      ->UseRealTime()
+
+#define BM_Eigen_Softmax(dtype, num_threads, label, ...)                       \
+  static void JOIN(BM_Eigen_Softmax_##dtype##_intraop_##num_threads##_dims,    \
+                   JOIN(CONCAT_DIMS, PP_NARG(__VA_ARGS__))(__VA_ARGS__))(      \
+      ::testing::benchmark::State & state) {                                   \
+    BM_Eigen_Softmax<dtype>(state, {__VA_ARGS__}, num_threads, label);         \
+  }                                                                            \
+  WRAP_BENCHMARK(JOIN(BM_Eigen_Softmax_##dtype##_intraop_##num_threads##_dims, \
+                      JOIN(CONCAT_DIMS, PP_NARG(__VA_ARGS__))(__VA_ARGS__)))   \
+      ->UseRealTime()
+
+// Benchmark with oneDNN library.
+BM_oneDNN_Softmax(float, 4, "float32_BERT_batch_size_1", 1, 16, 384, 384);
+BM_oneDNN_Softmax(float, 4, "float32_BERT_batch_size_16", 16, 16, 384, 384);
+BM_oneDNN_Softmax(float, 1, "float32_ImageNet_batch_size_32", 32, 1008);
+BM_oneDNN_Softmax(float, 1, "float32_ImageNet_batch_size_128", 128, 1008);
+BM_oneDNN_Softmax(float, 4, "float32_ImageNet_batch_size_32", 32, 1008);
+BM_oneDNN_Softmax(float, 4, "float32_ImageNet_batch_size_128", 128, 1008);
+BM_oneDNN_Softmax(bfloat16, 4, "bfloat16_BERT_batch_size_1", 1, 16, 384, 384);
+BM_oneDNN_Softmax(bfloat16, 4, "bfloat16_BERT_batch_size_16", 16, 16, 384, 384);
+BM_oneDNN_Softmax(bfloat16, 1, "bfloat16_ImageNet_batch_size_32", 32, 1008);
+BM_oneDNN_Softmax(bfloat16, 1, "bfloat16_ImageNet_batch_size_128", 128, 1008);
+BM_oneDNN_Softmax(bfloat16, 4, "bfloat16_ImageNet_batch_size_32", 32, 1008);
+BM_oneDNN_Softmax(bfloat16, 4, "bfloat16_ImageNet_batch_size_128", 128, 1008);
+
+// Benchamark with Eigen library.
+BM_Eigen_Softmax(float, 4, "float32_BERT_batch_size_1", 1, 16, 384, 384);
+BM_Eigen_Softmax(float, 4, "float32_BERT_batch_size_16", 16, 16, 384, 384);
+BM_Eigen_Softmax(float, 1, "float32_ImageNet_batch_size_32", 32, 1008);
+BM_Eigen_Softmax(float, 1, "float32_ImageNet_batch_size_128", 128, 1008);
+BM_Eigen_Softmax(float, 4, "float32_ImageNet_batch_size_32", 32, 1008);
+BM_Eigen_Softmax(float, 4, "float32_ImageNet_batch_size_128", 128, 1008);
+BM_Eigen_Softmax(bfloat16, 4, "bfloat16_BERT_batch_size_1", 1, 16, 384, 384);
+BM_Eigen_Softmax(bfloat16, 4, "bfloat16_BERT_batch_size_16", 16, 16, 384, 384);
+BM_Eigen_Softmax(bfloat16, 1, "bfloat16_ImageNet_batch_size_32", 32, 1008);
+BM_Eigen_Softmax(bfloat16, 1, "bfloat16_ImageNet_batch_size_128", 128, 1008);
+BM_Eigen_Softmax(bfloat16, 4, "bfloat16_ImageNet_batch_size_32", 32, 1008);
+BM_Eigen_Softmax(bfloat16, 4, "bfloat16_ImageNet_batch_size_128", 128, 1008);
```

Comment:
Nit: Please add an empty line after this line.

Reply:
Done.